src/NuGet.Clients/PackageManagement.VisualStudio/Migrate/ProjectJsonToPackageRefMigrator.cs

```diff
@@ -11,35 +11,50 @@
 using System.Linq;
 using System.Threading.Tasks;
 using Task = System.Threading.Tasks.Task;
+using Microsoft.VisualStudio.Threading;
 using NuGet.PackageManagement.UI;
+using Microsoft.VisualStudio.Shell;
 
 namespace NuGet.PackageManagement.VisualStudio
 {
     public static class ProjectJsonToPackageRefMigrator
     {
         public static async Task MigrateAsync(LegacyCSProjPackageReferenceProject project, string dteProjectFullName)
         {
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
yeah, i thought so too. but two things:
1) This is a public entry point.
2) If you are already on a UI thread this is a really fast operation, we aren't losing perf here.

Thoughts?